### DISCUSSION TITLE
Support for Huawei R4850G2 Alternativen?

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#236] [General]
I read that OPenDTUonBattery supports the Huawei R4850G2 power supply.

Will it also work with alternatives e.g. R4875G5 or R4850G5 ?

I am asking because the prices for the R4850G2 are very expensive.

**Comment by MalteSchm** [comment]
This is unclear. 
I think chances are not bad I have not heard of anybody who gave this a try

**Comment by madmartin** [comment]
There is one offer from the Netherlands for 95€ - what is "very expensive" for you?

https://www.kleinanzeigen.de/s-anzeige/power-supply-48v-3000w-r4850g2-huawei-for-rf-power-amplifier/2414724842-168-9520

**Reply by Manos1966** [reply]
This price is great!
Many Thanks for the tip, I am ordering!

**Reply by MalteSchm** [reply]
This is most likely the same guy that I got mine from

**Reply by Manos1966** [reply]
The offer from the Netherlands for 95€+12€ = 107€ is excellent.
Much better than the offers I have seen until now in the internet (160USD + 65USD delivery + customs?) from China!!!
Very annoying, considering they used to cost 50USD two years ago.

Anyway, I got a bit worried when I read it is a Kleinanzeigen German seller in Berlin but the item is in The Netherlands (I got scammed two times with some sellers and became more cautious).
I was relieved when I noticed they offer PayPal with insurance so, it sounds trustworthy.

In any case, I just bought my R4850G2 for 130EUR at eBay from this seller:
www.ebay.de/itm/325612924585
I happened to be in the area, I called the seller directly, went to his shop, he had a lot of them, they use them for radio transmitter applications, they put many of them in racks (picture included).  
![20230522_180031](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/1490f9d1-a922-40a7-852c-e297f48ed7bf)
![20230522_175959](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/124f1e7d-034d-47b0-9b72-d42294ffbd43)

Anyway, I picked one, he tested it in my presence and could take it home (please do not ask me what I will do with it... I just got excited reading about the options in the forum and decided to get one for testing, without having to pay a fortune for it...) 

IMPORTANT: Just in case someone wants to buy from this eBay link, the mentionned transportation cost of 44EUR is too expensive! Ask him for a better delivery price, I know that DPD charges 24,00 EUR for delivery to Germany (takes about one week) and since it is within the EU, there is no additional customs clearance charge.

In any case, this eBay seller is more expensive (130+24 = 154EUR) than the offer **madmartin** found in The Netherlands so, my first choice would be his suggestion!

**Comment by Manos1966** [comment]
Jetz habe ich noch eine Frage
Kommunikationsmodul(e) gibt es in zwei Versionen:
A- eine mit drei Anschluesse (L,N,G) fuer den 230V Eingang, und 
B- eine mit zwei (L,N) mit Schraube und ungewiss was sie mit dem G Anschluss gemacht haben, wahrscheinlich ist er auf der rechten Seite.

Ich haette lieber die B-Version um den 230V Anschluss sicher(er) anzuschliessen, bin ich aber nicht sicher ob eine Erdung existiert.
Hat jemand Erfahrung, kann jemand bestaetigen ob die Erdung bei der zweiten Version existiert?
![Stecker A   B](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/43a98b64-f727-490a-a5f7-d9342a8755a2)

**Reply by MalteSchm** [reply]
Ich meine ich habe Version B. Erde erfolgt da über den silbernen Kontakt ganz rechts. 
Die Erde ist tatsächlich der linke Kontakt von den drei Kontakten an dem Netzteil und da dann von rechts aussen über so eine Drahtkonstrution in die Mitte gezogen.

**Reply by ele-dev** [reply]
Ich habe auch die Version B. Also mein Multimeter hat angezeigt, dass der ganz äußere Kontakt für die Erdung und das Metall Gehäuse direkt miteinander verbunden sind. Das heißt du kannst das Netzteil alternativ auch Erden, indem du eine leitende Verbindung irgendwo auf der Gehäuse Oberfläche mit dem Schutzleiter herstellst.

**Reply by Manos1966** [reply]
Vielen Dank fuer die Informationen.

Ich habe die Version B bestellt und (Asche auf mein Haupt) das ganze liegt immer noch unbenutzt zu Hause.
Ich habe bis jetzt mit der Victron Loesung experimentiert.
Gestern kam auch das Joy-IT SBC-CAN01 Modul (vielen Dank @eu-gh fuer die WiKi Erklaerungen [https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU](url) )

**Jetzt aber eine Frage:**
Wie/woher nimmt OpenDTUonBattery die PV-Leistungsdaten, damit sie die Leistung des R4850G2 entsprechend einstellen kann?  
Vielleicht habe ich es verpasst, aber ich glaube keine Beschreibung gelesen zu haben,

**Reply by eu-gh** [reply]
Version A oder B ist im Prinzip doch egal, solange man alles korrekt verdrahtet.

Die automatische interne Leistungsregelung setzt eine aktive Energiemesseinrichtung irgendwo im Haushalt voraus, welche ihre Daten über MQTT veröffentlicht. Der aus meiner Sicht einfachste Fall ist die Nutzung eines IR-Lesekopfs für den neuen Stromzähler, manche Leute bauen sich auch eine Shelly in ihren Sicherungskasten. Dessen Topic trägst du dann bei [Einstellungen/Power Meter] ein und danach funktioniert das (relativ) eigenständig.

**Reply by Manos1966** [reply]
Hallo @eu-gh 
im Prinzip vielleicht, aber lieber habe ich zwei 230V Drähte mit Schrauben fixiert, als mit den Steckern der Version A. 

> die Nutzung eines IR-Lesekopfs für den neuen Stromzähler, 

Jetzt bin ich aber total verwirrt! 
Der Hausstromverbrauch wird benutzt um die Hoymiles Produktion zu steuern.
Wozu dienen diese Werte für den R4850G2? Er lädt die Batterie, also muss er wissen, wie viel Strom die PV Anlage produziert. 


Hmm... vielleicht sollte ich die Frage präziser stellen: Welche Parameter wird benutzt um die Leistung des R4850G2 einzustellen? z.B. wenn die PV Anlage 500W proziert, dann wird der R4859G2 auf 500W gestellt (solange die Batterie den max Ladestand % nicht erreicht hat).

**Reply by eu-gh** [reply]
Okay, also macht deine PV aktuell bereits eine aktive Nulleinspeisung? Ansonsten müsste der Hauszähler ja bei erzeugtem>genutztem Strom einen negativen Momentanverbrauch anzeigen, den das Huawei-NT dann als Basis für die Leistungsregelung nutzen könnte. Das werden nie 100% vom Ausgangswert der PV sein, weil man ja immer irgendwelche Verbraucher hat.

**Reply by Manos1966** [reply]
@eu-gh ich bin nicht sicher ich verstehe das, dass du da schreibst....

Die dynamische Nulleinspeisung regelt die Hoymiles so, dass der Stromzeller Null zeigt (Nulleinspeisung eben 😂).

Jetzt erklär mir bitte wie ein Wert von Null helfen wird, die R4850G2 so einzustellen,  damit die Batterie geladen wird.

Ich wiederhole meine Frage: Welche Parameter wird benutzt um die Leistung des R4850G2 einzustellen? z.B. wenn die PV Anlage 500W proziert, dann wird der R4859G2 auf 500W gestellt (solange die Batterie den max Ladestand % nicht erreicht hat).

**Reply by eu-gh** [reply]
Bis eben war (zumindest mir) ja noch nicht klar, dass deine PV Nulleinspeisung macht. Das ist nicht der Standardfall und relativer Käse, wenn man ein externes Batteriemanagement hat, weil der Strom dann ja eingangsseitig gar nicht erst erzeugt wird.

Du bräuchtest also eine (externe) Logik, die in Echtzeit aus deiner ersten oDTU ausliest, was die PV aktuell erzeugen KÖNNTE (z.B. 800W), dann den aktuellen Ertrag (=Verbrauch wg. Nulleinspeisung, z.B. 500W) davon abzieht und diesen Wert (in diesem Beispiel folglich 300W) laufend an die oDTUoB schickt, damit genau das dann in die Batterie geladen wird. Mit den Bordmitteln dieses Projektes alleine wirst du das nicht schaffen, glaube ich.

**Comment by MalteSchm** [comment]
> @eu-gh ich bin nicht sicher ich verstehe das, dass du da schreibst....
> 
> Die dynamische Nulleinspeisung regelt die Hoymiles so, dass der Stromzeller Null zeigt (Nulleinspeisung eben 😂).
> 
> Jetzt erklär mir bitte wie ein Wert von Null helfen wird, die R4850G2 so einzustellen, damit die Batterie geladen wird.
> 
> Ich wiederhole meine Frage: Welche Parameter wird benutzt um die Leistung des R4850G2 einzustellen? z.B. wenn die PV Anlage 500W proziert, dann wird der R4859G2 auf 500W gestellt (solange die Batterie den max Ladestand % nicht erreicht hat).

Das ganze dient dazu Nulleinspeisung mit einer Anlage die über AC gekoppelt ist und damit die Batterie nicht direkt laden kann zu erreichen. Außerdem kann man dynamische Stromtarife nutzen und dann laden wenn günstig.

**Comment by Manos1966** [comment]
@MalteSchm @eu-gh 
OK, ich musste einiges lesen um durchzublicken ('was ich nicht behaupte, erreicht zu haben...), aber im Prinzip, die Loesung ist einfacher als ich dachte:

**Wenn (falls) ich es richtig verstanden habe,** die OpenDTUonBattery hat zur Zeit:
- Steuerung des Huawei  R4859G2
- Dynamische Anpassung dessen Ladeleistung auf Basis des PowerMeters-Daten

Ich brauche also:
- eine ZWEITE OpenDTUonBattery
- Ein Messgeraet (Shelly1PM oder aehnliches) direkt nach der Strom produzierenden HOYMILES (sie liefert die 230V Strom-Produktion der PV Paneele)
-  Diese Daten werden in PowerMeter definiert (und NICHT die Messdaten vom Stromzaehler da Nulleinspeisung)

Gleichzeitig die ERSTE OpenDTUonBattery kuemmert sich um die Nulleinspeisung und steuert die andere HOYMILES die an die Pylotech angeschlossen ist und die Daten des Stromzaehlers benutzt.

Klingt eigentlich nicht schlecht (wenn es stimmt), die extra-Kosten einer zweiten OpenDTUonBattery lassen sich ja verkraften :-d 

ABER dann bleibt eine letzte Frage:
Die Pylontech Batterie muss and OpenDTUonBat EINS und ZWEI angeschlossen sein. Ist es moeglich?

**Comment by MalteSchm** [comment]
openDTU kann das gleichzeitig. Du brauchst keinen zweiten ESP
Vorausgesetzt du hast einen HM xxxx Inverter mit dem NRF Chip und nicht den anderen mit WLAN

Pylontech, Hoymiles, der Victron und das Netzteil gehen alle am gleichen ESP32

Und was zum Hausverbrauch messen braucht es auch. Shelly, Tasmota, oder so

**Reply by Manos1966** [reply]
Moment Mal @MalteSchm 

> openDTU kann das gleichzeitig. Du brauchst keinen zweiten ESP

Das ist die Frage die ich staendig stelle und keiner hat mir bis jetzt eine Antwort gegeben:
**Wie wird der R4850G2 gesteuert?** 
Welche Parameter wird benutzt um die Leistung des R4850G2 einzustellen? 
z.B. wenn die PV Anlage 500W produziert, dann soll der R4859G2 auf 500W gestellt (solange die Batterie den max Ladestand % nicht erreicht hat).

Denn so lange der eine Hoymiles (der an die Batterie angeschlossen ist) Strom produziert, habe ich eine Nulleinspeisung am Stromzaehler. Perfekt! 👍 
Woher wird also die OpenDTUonBAt wissen, dass (vom zweiten Hoymiles) Strom produziert wird um dann den  R4859G2 auf die gleiche Leistung einzustellen?

**Comment by MalteSchm** [comment]
Wo stellst du diese Frage?
Wenn du im Haus überschuss hast wird das Netzteil so geregelt das damit die Batterie geladen wird. Diese Info bekommt openDTU vom Shelly oder Tasmota etc...

Haus braucht Strom: Batterie wird entladen
Haus erzeugt Strom: Batterie wird über Netzteil geladen

Was ich nicht weiß ist wie gut das mit richtig kleinen Leistungen klappt. Ich lasse das gerade von 150-2000W laufen

**Reply by Manos1966** [reply]
Leute ihr macht mich verrueckt!

OK @MalteSchm , hast du ueberhaupt einen R4850G2 im Einsatz? 
Dann poste mal bitte deine **Dynamic Power Limiter Settings** 
Ich bin gespannt zu verstehen woher die OpenDTU erkennt dein "Haus braucht Strom" / "Haus erzeugt Strom" wenn die Aufgabe des Dynamic Power Limiter, eine Nulleinspeisung zu halten ist.

**Comment by MalteSchm** [comment]
Ja, ist hier nachzulesen

https://www.akkudoktor.net/forum/stell-dein-batterie-powerwall-projekt-vor/ac-dc-speicherloesung-mit-victron-mppt-pylontech-hoymiles-huawei-und-opendtu-onbattery/

Wenn vom Power Meter ein positiver Wert kommt liegt Verbrauch vor, dann springt der Power Limiter an um regelt den Hoymiles.
Kommt ein negativer Wert ist überschuss vorhanden und das Netzteil wird geregelt. Das macht aber der Code den ich mal für das Netzteil geschrieben habe

**Comment by Manos1966** [comment]
Oh ich sehe, ich war auf dem falschen Pferd mit meiner Gedanken.
Natuerlich funktioniert es mit den PowerMeter-Daten 👍 
Ich habe unterdrueckt (=vergessen), dass der Dynamic Power Limiter nur einen HOYMILES steuert.

Vilen Dank @MalteSchm und @eu-gh

**Comment by Manos1966** [comment]
@MalteSchm welchen Kasten hast du benutzt und wie hast du das Netzteil befestigt?
![16519-PXL20230326184934165-MP - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/c2892b86-8123-484d-8267-28d59d9af704)

**Reply by MalteSchm** [reply]
Das ist ein 3D Druckteil. Es trägt den Stecker und dient als Luftkanal

Im Grunde hält das Netzteil durch den Stecker. Der schwarze Hebel (links im Bild zu sehen/erahnen) würde das aber auch tun sobald es um 1mm raus rutschen würde

Nimm auf jeden Fall einen größeren Schaltschrank als ich.

**Reply by Manos1966** [reply]
Danke @MalteSchm 
stoert dich deinen 3D Print zu teilen?

> Im Grunde hält das Netzteil durch den Stecker. 

Mein Stecker (Version B) kann nicht ans Netzteil festgeschraubt werden (ich habe keine Loecher gesehen). 
Wie hast du es gemacht?

Und wie waere es mit einem Photo von dieser Seite?
Hast du Luftungsloecher auf dieser Seite?
![16519-PXL20230326184934165-MP - Copy (2)](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/78e12537-49dc-457f-8d39-a778b3fc8d56)

**Reply by MalteSchm** [reply]
> Danke @MalteSchm stoert dich deinen 3D Print zu teilen?
> 
> > Im Grunde hält das Netzteil durch den Stecker.
> 
> Mein Stecker (Version B) kann nicht ans Netzteil festgeschraubt werden (ich habe keine Loecher gesehen). Wie hast du es gemacht?

Mein Stecker ist mit dem Druckteil verschraubt. Keine Ahnung ob ich den mit dem Netzteil verschrauben könnte. Finde ich aber blöd weil das ja in dem Luftkanal passieren müsste. Das war offenbar nicht klar genug geschrieben: Es hält bei mir vor allem durch die Klemmung der Stecker.

Druckteil anbei:
[huawei_netzteil_halter.zip](https://github.com/helgeerbe/OpenDTU-OnBattery/files/12735333/huawei_netzteil_halter.zip)

> Und wie waere es mit einem Photo von dieser Seite? Hast du Luftungsloecher auf dieser Seite?

Sorry habe ich kein Bild von gemacht. Aber klar sind da Löcher für die Luft. Da sitzen zwei Lüftungsgitter vor (60x60mm wenn ich es richtig im Kopf habe) gibts z.B. bei ebay.

**Reply by Manos1966** [reply]
Vielen Dank!

**Reply by Manos1966** [reply]
Danke fuer die Infos @MalteSchm, jetzt wird endlich mein 3D Drucker etwas sinnvolles zu tun. 
welches Material hast du benutzt PLA oder ABS? Ich habe Angst, das PLA mit der Hitze kaputt gehen wird.

Diese hast du auch selber gedruckt?
![16523-PXL20230326184939534 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/0334f7ea-b246-44c4-b1fd-5d15d77d4b6e)

**Reply by MalteSchm** [reply]
Ja auch selbst gedruckt. Diese Datei finde ich aber nicht mehr. Das hält aber nur das Kabel da
Ist PLA. Bisher hält es. Ich denke das das geht.

**Reply by Manos1966** [reply]
OK, dann PLA! Ich fange an zu drucken...
![WhatsApp Image 2023-09-28 at 16 22 46](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/058a0326-3a11-4581-8bce-fcf85583e658)

**Reply by Manos1966** [reply]
@MalteSchm Sorry dass ich dich nerve, aber ist dieser Spalt gewollt?
![Detail Halter](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/c1d8206c-847c-4a1c-aa93-23cdda5f7f20)

**Reply by Manos1966** [reply]
@MalteSchm I hoffe du hast nichts dagegen wenn ich die 3D Datei in die Wiki addiere...

**Reply by MalteSchm** [reply]
Mach ruhig

**Reply by Manos1966** [reply]
Ist schon drin:
[https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-R4850G2-power-supply-solution-(Example)](url)

Hast du vielleicht dein 3D Model in FreeCAD, oder aehnliche Freeware erstellt?
Weil jetzt dass ich einen grosseren Schaltkasten bestellt habe, muss ich das Ende verlaengen
![Capture2 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/8345ff5d-53ec-4182-888f-ec48cc94a828)

**Reply by MalteSchm** [reply]
Nein. Aber deshalb ist da die STP Datei in dem ZIP file

**Comment by Manos1966** [comment]
@MalteSchm Jetzt verstehe ich die "Genialitaet" deines 3D-Models 👍 
![20231005_113214 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/6a2594c1-5442-4fa8-b295-23a0fd130c2f)

Du hast nicht erwaehnt, wie du das Netzteil auf der anderen Seit befestigt hast (damit es nicht rauskommt)
![20231005_113207 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/f2ec421f-6a10-49c8-828f-ea72ec795580)

Ich muss jetzt lernen, wie ich dein 3D Model "aufsplitten" kann, damit ich die notwendige Verlaengerung addieren kann... bis jetzt hat es mit FreeCAD und OpenSCAD nicht geklappt, der Kampf geht weiter :-d
